abci/src/application.rs

```diff
@@ -126,53 +91,3 @@ pub trait Application {
         Default::default()
     }
 }
-
-/// Provides a mechanism for the [`Server`] to execute incoming requests while
-/// expecting the correct response types.
-pub trait RequestDispatcher {
-    /// Executes the relevant application method based on the type of the
-    /// request, and produces the corresponding response.
-    fn handle(&self, request: Request) -> Response;
```

Comment:
other solutions that come to mind are not fundamentally different, but what definitely could be improved is to use different type for this kind of status code (continue to receive TD messages or not), because `Result` carries an effect of possibly failing computation and adding a special error variant to indicate expected end of communication looks hacky